<?xml version="1.0"?>

<!--
 this version of tracking.launch has master set as false by default for use
 while snav or modalai vision lib is running since snav is master.
 Also snav_raw10_compatability flag
 is set to true which allows downsampling the 10-bit images snav uses into
 8-bit mono images that can be used by ROS
-->

<launch>
    <!-- cmd line arguments to launch file -->
    <arg name="cam_name" default="tracking"/> <!-- sets the topic namespace -->
    <arg name="cam_id" default="$(env TRACKING_CAM_ID)"/> <!-- configure with voxl-configure-cameras -->
    <arg name="skip_n_frames" default="0"/>
    <arg name="frame_rate" default="15" />
    <arg name="width" default="640"/>
    <arg name="height" default="480"/>
    <arg name="frame_id" default="tracking_frame"/> <!-- transform frame used by TF2 -->
    <arg name="do_rectification" default="false"/>

    <!-- start node(s) -->
    <group ns="$(arg cam_name)">
        <node pkg="nodelet" type="nodelet" name="$(arg cam_name)_nodelet_manager"
            args="manager" output="screen"/>

        <!-- camera driver nodelet -->
        <node pkg="nodelet" type="nodelet" name="voxl_cam_nodelet"
            args="load voxl_cam_ros/driver $(arg cam_name)_nodelet_manager">
            <param name="cam_name" value="$(arg cam_name)"/>
            <param name="cam_id" value="$(arg cam_id)"/>
            <param name="skip_n_frames" value="$(arg skip_n_frames)"/>
            <param name="frame_rate" value="$(arg frame_rate)" />
            <param name="width" value="$(arg width)"/>
            <param name="height" value="$(arg height)"/>
            <param name="frame_id" value="$(arg frame_id)"/>
            <param name="is_cam_master" value="false"/>
            <param name="snav_raw10_compatability" value="true"/>
        </node>

        <!-- mono rectification -->
        <group if="$(arg do_rectification)">
            <node pkg="nodelet" type="nodelet" name="image_proc_rect"
            args="load image_proc/rectify $(arg cam_name)_nodelet_manager">
                <remap from="image_mono" to="image_raw"/>
            </node>
        </group>
    </group>

</launch>
